src/main/network/router/codel_queue.rs

```diff
@@ -111,20 +320,22 @@ impl CoDelQueue {
     }
 
     fn drop(&self, packet: Packet) {
-        // packet_addDeliveryStatus(packet, PDS_ROUTER_DROPPED);
-        // #ifdef DEBUG
-        //     gchar* pString = packet_toString(packet);
-        //     trace("Router dropped packet %s", pString);
-        //     g_free(pString);
-        // #endif
-        //     packet_unref(packet);
+        packet.add_status(PacketStatus::RouterDropped);
+        drop(packet);
```

Comment:
You don't need a `drop` here, it will be dropped automatically when it goes out of scope.